src/testers.jl

```diff
@@ -112,61 +112,55 @@ at input point `z` to confirm that there are correct `frule` and `rrule`s provid
 
 `fkwargs` are passed to `f` as keyword arguments.
 All keyword arguments except for `fdm` and `fkwargs` are passed to `isapprox`.
+
+To use this tester for a scalar type `MyNumber <: AbstractNumber`,
+`FiniteDifferences.to_vec(::MyNumber)` must be implemented.
 """
 function test_scalar(f, z; rtol=1e-9, atol=1e-9, fdm=_fdm, fkwargs=NamedTuple(), kwargs...)
     _ensure_not_running_on_functor(f, "test_scalar")
-    # z = x + im * y
-    # Ω = u(x, y) + im * v(x, y)
     Ω = f(z; fkwargs...)
 
+    vz, z_from_vec = to_vec(z)
+    # orthonormal tangent vectors
+    vz_basis = Diagonal(ones(eltype(vz), length(vz)))
+    Δzs = [z_from_vec(vz_basis[:, i]) for i in axes(vz_basis, 2)]
```

Comment:
alternatively would it be clearer to as `onehot` function?
```suggestion
    function onehot(i)
        v=zeros(eltype(vz), length(vz))
        v[i] = one(eltype(vz))
        return v
     end
    Δzs = [z_from_vec(onehot(i)) for i in axes(vz_basis, 2)]
```

Alternatively would it be ok to use views here?
```suggestion
    vz_basis = Diagonal(ones(eltype(vz), length(vz)))
    Δzs = [z_from_vec(@view vz_basis[:, i]) for i in axes(vz_basis, 2)]
```


idk.
to my surprise 
`vz_basis[:, i]`  is a `SparseVector` 
 so maybe that is the ideal type to be using anyway?